specification/sbi_cove.adoc

```diff
@@ -1651,6 +1677,65 @@ COVE-Guest calls cause a trap to the TSM. TSM should do any processing required
 and then must forward the ECALL to the host with `scause` set to ECALL, `a7` set
 to EID, `a6` set to FID (See <<cove-fid>>), `a0`-`a5` set to ECALL args.
 
+[#sbi_covg_promote_to_tvm]
+=== Function: COVE Guest Promote to TVM (FID #0)
+[source, C]
+-----
+struct sbiret sbi_covg_promote_to_tvm(unsigned long fdt_addr,
+                                      unsigned long lap_addr);
+-----
+A VM requests a TSM to be promoted to a TVM. The `fdt_addr` is the guest physical address 
+of the flattened device tree (FDT). `lap_addr` is the guest physical address of the 
+`local attestation payload` used for `local attestation`. VMs that do not require 
+local attestation (only the remote attestation) must write `0` to `lap_addr`.
+
+The VM must make this call early during the boot process, so that the content of memory
+is known and the resulting integrity measurement hash is deterministic. TSM uses VM page table to copy
+VM's data to confidential memory. TSM extracts from FDT information about the number of vCPUs, allocating
+required amount of memory and populating one with the vCPU state of the caller.
+
+On success, TSM reflects an SBI call `sbi_register_tvm(unsigned long tvm_guest_id)` to the hypervisor.
+The hypervisor marks the VM as TVM and releases the VM memory. For now on, OS/VMM interacts with TVM
+with help of TSM using the COVH ABI. For example, it resumes TVM using the `sbi_covh_run_tvm_vcpu()` call.
+
+If the call fails, the TSM returns SBI error code in `sbiret.error` to the VM. The possible error codes 
+are shown below.
+
+[#table_sbi_covg_promote_to_tvm_errors]
+.COVE Guest Promote to TVM Errors
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_INVALID_ADDRESS | `fdt_addr` was invalid.
+| SBI_ERR_AUTH            | Local attestation failed.
+| SBI_ERR_OUT_OF_MEMORY   | Not enough memory to store TVM
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+[#covg_retrieve_secret]
+=== Function: COVE Guest Retrieve Secret (FID #0)
```

Comment:
FID #1?

Reply:
I adjusted FIDs to be unique in COVG after the addition of new calls.